AppCenter/AppCenterTests/MSHttpClientTests.m

```diff
@@ -0,0 +1,321 @@
+                               }];
+
+  // Then
+  XCTAssertEqualObjects(actualRequest.URL, url);
+  XCTAssertEqualObjects(actualRequest.HTTPMethod, method);
+  XCTAssertEqualObjects(actualRequest.allHTTPHeaderFields, headers);
+}
+
+- (void)testDeleteUnrecoverableErrorWithoutHeadersNotRetried {
+
+  // If
+  // Stub HTTP response.
+  __block int numRequests = 0;
+  __block NSURLRequest *actualRequest;
+  [OHHTTPStubs
+      stubRequestsPassingTest:^BOOL(__attribute__((unused)) NSURLRequest *request) {
+        ++numRequests;
+        actualRequest = request;
+        return YES;
+      }
+      withStubResponse:^OHHTTPStubsResponse *(__attribute__((unused)) NSURLRequest *request) {
+        return [OHHTTPStubsResponse responseWithData:[NSData data] statusCode:MSHTTPCodesNo400BadRequest headers:nil];
+      }];
+  __weak XCTestExpectation *expectation = [self expectationWithDescription:@""];
+  self.sut = [MSHttpClient new];
+  NSURL *url = [NSURL URLWithString:@"https://mock/something?a=b"];
+  NSString *method = @"DELETE";
+  NSDictionary *headers = [NSDictionary new];
+  NSData *payload = nil;
+
+  // When
+  [self.sut sendAsync:url
+                 method:method
+                headers:headers
+                   data:payload
+      completionHandler:^(NSData *responseBody, NSHTTPURLResponse *response, NSError *error) {
+        // Then
+        XCTAssertEqual(response.statusCode, MSHTTPCodesNo400BadRequest);
+        XCTAssertNotNil(responseBody);
+        XCTAssertNil(error);
+        [expectation fulfill];
+      }];
+
+  [self waitForExpectationsWithTimeout:kMSTestTimeout
+                               handler:^(NSError *_Nullable error) {
+                                 if (error) {
+                                   XCTFail(@"Expectation Failed with error: %@", error);
+                                 }
+                               }];
+  // Then
+  XCTAssertEqualObjects(actualRequest.URL, url);
+  XCTAssertEqualObjects(actualRequest.HTTPMethod, method);
+  XCTAssertEqualObjects(actualRequest.allHTTPHeaderFields, headers);
+  XCTAssertEqual(numRequests, 1);
+}
+
+- (void)testNetworkDownAndThenUpAgain {
+
+  // If
+  // Stub HTTP response.
+  __block int numRequests = 0;
+  __block NSURLRequest *actualRequest;
+  __block BOOL completionHandlerCalled = NO;
+  __weak XCTestExpectation *expectation = [self expectationWithDescription:@""];
+  dispatch_semaphore_t timingSemaphore = dispatch_semaphore_create(0);
+
+  [OHHTTPStubs
+      stubRequestsPassingTest:^BOOL(__attribute__((unused)) NSURLRequest *request) {
+        ++numRequests;
+        actualRequest = request;
+        return YES;
+      }
+      withStubResponse:^OHHTTPStubsResponse *(__attribute__((unused)) NSURLRequest *request) {
+        return [OHHTTPStubsResponse responseWithData:[NSData data] statusCode:MSHTTPCodesNo204NoContent headers:nil];
+      }];
+  self.sut = [[MSHttpClient alloc] initWithRetryIntervals:@[ @1 ] reachability:self.reachabilityMock];
```

Comment:
All other retry intervals are doubles (or whatever the default is for numbers with fractional parts), any reason this one is `@1` instead of `@1.0`?

Reply:
AppCode refactor inline did that. Will restore.

Reply:
Actually should be ints.

Reply:
Removing `.0` instead where used.